Mathlib/RingTheory/AlgebraicIndependent/TranscendenceBasis.lean

```diff
@@ -540,3 +540,34 @@ variable (R S A)
     trdeg R S + trdeg S A = trdeg R A := by
   rw [← (trdeg R S).lift_id, ← (trdeg S A).lift_id, ← (trdeg R A).lift_id]
   exact lift_trdeg_add_eq R S A
+
+variable {R S} in
+/-- If `t ∪ {j}` is a transcendence basis and `j` is algebraic over `t ∪ {i}`, then `t ∪ {i}` is
+also a transcendence basis. -/
+lemma IsTranscendenceBasis.of_isAlgebraic_adjoin_image_compl
+    [FaithfulSMul R S] [NoZeroDivisors S] (i j : ι) (v : ι → S)
+    (H₁ : IsTranscendenceBasis R fun x : {x // x ≠ i} ↦ v x)
+    (H₂ : IsAlgebraic (Algebra.adjoin R (v '' {j}ᶜ)) (v j)) :
+    IsTranscendenceBasis R fun x : {x // x ≠ j} ↦ v x := by
```

Comment:
This could be easily generalized to
```suggestion
lemma IsTranscendenceBasis.of_isAlgebraic_adjoin_image_compl
    [FaithfulSMul R S] [NoZeroDivisors S] (s : Set ι) (i j : ι) (v : ι → S)
    (H₁ : AlgebraicIndepOn R v s) (H₂ : ¬ AlgebraicIndepOn R v (insert i s))
    (alg : IsAlgebraic (Algebra.adjoin R (v '' (insert i s \ {j}))) (v j)) :
    AlgebraicIndepOn R v (insert i s \ {j}) := by
```

Reply:
In fact I'll extract both statements to matroid lemmas.

Reply:
#26510